From 36232ff0b12c42bc4654ab3774a5fce79e544cac Mon Sep 17 00:00:00 2001
From: Philippe Coval <p.coval@samsung.com>
Date: Mon, 23 Jul 2018 10:58:11 +0200
Subject: [PATCH] tizenrt: Disable error on warning temporary

This is needed to build on nettest image,
or rebase on TizenRT tag 2.0_GBM_M1

Change-Id: I6b59d5518f874a516b2d72453030853c1c2202ab
Bug: https://jira.iotivity.org/projects/LITE/issues/LITE-4
Signed-off-by: Philippe Coval <p.coval@samsung.com>
---
 tests/port/tizenrt/Makefile | 21 +++++++++++++++++----
 1 file changed, 17 insertions(+), 4 deletions(-)

diff --git a/tests/port/tizenrt/Makefile b/tests/port/tizenrt/Makefile
index 31197da5..35f06e61 100644
--- a/tests/port/tizenrt/Makefile
+++ b/tests/port/tizenrt/Makefile
@@ -52,6 +52,7 @@ PATH:=${toolchain_bin}:${PATH}
 export PATH
 
 prep_files+=${CC}
+prep_files+=${tizenrt_dir}/tmp/done.log
 prep_files+=${tizenrt_dir}/external/iotivity-constrained.Kconfig.tmp
 prep_files+=${tizenrt_dir}/external/iotivity-constrained.mk.tmp
 prep_files+=${top_dir}/deps/tinycbor/LICENSE
@@ -59,7 +60,6 @@ prep_files+=${top_dir}/deps/tinycbor/tmp/done.log
 
 all?=prep configure build
 
-
 ${tizenrt_dir}:
 	git clone --depth 1 --recursive -b "${os_branch}" "${os_url}" "$@"
 
@@ -96,20 +96,33 @@ export IOTIVITY_CONSTRAINED_BASE_DIR
 #${tizenrt_dir}/external/iotivity-constrained:
 #	ls $@ || ln -fs ${CURDIR}/../.. $@
 
-build: ${os_dir}/.config
+build: ${tizenrt_dir}/build/output/bin/tinyara_head.bin
+	ls $<
+
+${tizenrt_dir}/build/output/bin/tinyara_head.bin: ${os_dir}/.config
 	${MAKE} -C "${<D}"
+	ls -l $@
 
-${configure_file}: ${tizenrt_dir} prep
+${configure_file}: ${tizenrt_dir} ${CC}
 	ls "$@" || ${MAKE} rule/configure
 
 configure: ${configure_file}
 	ls "$<"
 
-rule/configure: ${tizenrt_dir}/os/tools
+rule/configure: ${tizenrt_dir}/os/Make.defs
+	ls "$<"
+
+${tizenrt_dir}/os/Make.defs: ${tizenrt_dir}/os/tools ${CC}
 	cd "$<" && ./configure.sh ${os_machine}/${os_image_type}
 	echo "CONFIG_ENABLE_IOTIVITY_CONSTRAINED=y" >> ${configure_file}
 	echo "CONFIG_IOTLITE_EASYSETUP=y" >> ${configure_file}
 
+${tizenrt_dir}/tmp/done.log: ${tizenrt_dir}/os/Make.defs $(wildcard ${tizenrt_dir}/build/configs/*/*/Make.defs)
+	sed -e 's/-Werror//g' -i $^
+	mkdir -p ${@D}
+	echo "Patches applied in $^" > "$@"
+	echo "TODO: Enable Werror again on next TizenRT update"
+
 prep: ${prep_files}
 	ls $^
 
-- 
2.16.1.windows.1

